content/ui-patterns/loading.mdx

```diff
@@ -0,0 +1,406 @@
+
+</Box>
+
+<img width="456" alt="A progress bar with % completed text" src="https://github.com/primer/design/assets/2313998/973ab0ca-4f5c-4e06-98f2-e66fdf706943" />
+</Box>
+
+### Indeterminate loading indicators
+
+<div>
+
+Indeterminate loading indicators may be used when the progress or duration of a process is variable or unknown.
+
+This type of loading gives the user a sense that something is happening, but not how long it will take.
+
+</div>
+
+<Box display="flex" flexDirection={['column', 'column', 'column', 'column', 'row']} sx={{alignItems: 'flex-start', gap: 3}}>
+
+<div>
+
+#### Spinner
+
+A [spinner](/ui-packages/spinner) is the most versatile indeterminate loading indicator. It may be used in place of content that is loading, or adjacent to a loading message
+
+</div>
+
+<img width="456" alt="PR status checks being replaced by a spinner" src="https://github.com/primer/design/assets/2313998/4f4e897e-bbda-4cbc-bd0f-962450c6b5b6" />
+</Box>
+
+<Box display="flex" flexDirection={['column', 'column', 'column', 'column', 'row']} sx={{gap: 3}}>
+
+<Box flex={1}>
+
+#### Content skeleton
+
+For large areas of loading content, the loading content may be replaced by a vague representation of the content. This shows users the general shape of the page and could make them perceive the real content as loading faster.
+
+Skeleton loaders may be used when we can approximate the visual shape the loaded content will create on the page. It would feel weird if we showed a content skeleton and then the loaded content looks drastically different.
+
+The [tree view](/components/tree-view) and [data table](/components/data-table) components have these states implemented.
+
+</Box>
+
+<CustomVideoPlayer
+  width="456"
+  loop
+  autoplay
+  src="https://github.com/primer/react/assets/2313998/4bb7b7b6-482f-46c8-97d8-4b560066d287"
+/>
+</Box>
+
+<Box display="flex" flexDirection={['column', 'column', 'column', 'column', 'row']} sx={{gap: 3}}>
+
+<Box flex={1}>
+
+#### Animated icon or illustration
+
+<>
+{/*<!-- _TODO: reconsider if we really want to endorse this by mentioning it in Primer docs_ -->*/}
+</>
+
+An animated icon or illustration may be used as a "branding moment" for special loading states. This should be used sparingly.
+
+</Box>
+
+<CustomVideoPlayer
+  width="456"
+  loop
+  autoplay
+  src="https://github.com/primer/react/assets/2313998/d5a49e2d-315c-4d55-b929-ef2e6a648712"
+/>
+</Box>
+</Box>
+
+#### Loading text
+
+Text may accompany any of the above loading indicators to give the user more context about the process.
+
+## Adapting to different wait times
+
+The following is meant as general guidance since we can't accurately predict how long a process will take. It's difficult to give a even a rough estimate how long a process will take due to unknown variables such as network speed, device performance, and server load.
```

Comment:
```suggestion
The following is meant as general guidance since we can't accurately predict how long a process will take. It's difficult to give even a rough estimate of how long a process will take due to unknown variables such as network speed, device performance, and server load.
```